docs/source/public_server.rst

```diff
@@ -23,56 +29,96 @@ Securing a notebook server
 --------------------------
 
 You can protect your notebook server with a simple single password by
-setting the :attr:`NotebookApp.password` configurable. You can prepare a
-hashed password using the function :func:`notebook.auth.security.passwd`:
+configuring the :attr:`NotebookApp.password` setting in
+:file:`jupyter_notebook_config.py`.
+
+Prerequisite: A notebook configuration file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Check to see if you have a notebook configuration file,
+:file:`jupyter_notebook_config.py`. The default location for this file
+is your Jupyter folder in your home directory, ``~/.jupyter``.
+
+If you don't already have one, create a config file for the notebook
+using the following command::
+
+  $ jupyter notebook --generate-config
+
+
+Preparing a hashed password
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can prepare a hashed password using the function
+:func:`notebook.auth.security.passwd`:
 
 .. sourcecode:: ipython
 
     In [1]: from notebook.auth import passwd
     In [2]: passwd()
-    Enter password: 
-    Verify password: 
+    Enter password:
+    Verify password:
     Out[2]: 'sha1:67c9e60bb8b6:9ffede0825894254b2e042ea597d771089e11aed'
-    
-.. note::
 
-  :func:`~notebook.auth.security.passwd` can also take the password as a string
-  argument. **Do not** pass it as an argument inside an IPython session, as it
+.. caution::
+
+  :func:`~notebook.auth.security.passwd` when called with no arguments
+  will prompt you to enter and verify your password such as
+  in the above code snippet. Although the function can also
+  be passed a string as an argument such as ``passwd('mypassword')``, please
+  **do not** pass a string as an argument inside an IPython session, as it
   will be saved in your input history.
 
-You can then add this to your :file:`jupyter_notebook_config.py`, e.g.::
+Adding hashed password to your notebook configuration file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can then add the hashed password to your :file:`jupyter_notebook_config.py`.
+The default location for this file :file:`jupyter_notebook_config.py` is in
+your Jupyter folder in your home directory, ``~/.jupyter``, e.g.::
 
-    # Password to use for web authentication
+    # Get notebook configuration and add hashed password
     c = get_config()
-    c.NotebookApp.password = 
+    c.NotebookApp.password =
     u'sha1:67c9e60bb8b6:9ffede0825894254b2e042ea597d771089e11aed'
```

Comment:
can you put the hash on the same line just after the `=` ? I think it will be better.